test/Test_Optional.cpp

```diff
@@ -0,0 +1,41 @@
+//
+// Purpur Tentakel
+// cpp 23
+// 12.05.24
+//
+
+
+#include <cpt/Optional.hpp>
+#include <gtest/gtest.h>
+
+enum class Entry {
+    default_entry,
+    other_entry,
+};
+
+TEST(Optional, OK) {
+    auto const optional = []() -> cpt::Otional<Entry> { return Entry::default_entry; }();
```

Comment:
Why do you use an IIFE instead of just `auto const optional = cpt::Optional<Entry>{ Entry::default_entry };`?

Reply:
I use it to make sure a return will pick the right constructor.
And that it will work implicit.
because that will be the main use case.
Does that make sense?

Reply:
I don't understand what you mean 🤔

Reply:
i want to make sure, then no one can mark the constructors as `explicit`.
your version would still work.
but my would be broken.

or do I muss something? 😅

Reply:
tried something different.
that should work.
discussion with r00tifant on DC. 😄